tasktiger/task.py

```diff
@@ -286,30 +298,38 @@ def _move(self, from_state=None, to_state=None, when=None, mode=None):
                 # TODO: Do the following two in one call.
 
                 # Delete executions if there were no errors.
-                if from_state == ERROR:
-                    check_states = {}
-                else:
-                    check_states = {ERROR}
-                scripts.delete_if_not_in_zsets(
-                    _key("task", self.id, "executions"),
-                    self.id,
-                    [_key(state, queue) for state in check_states],
-                    client=pipeline,
-                )
+                if from_state != ERROR:
```

Comment:
Just realised that we should probably just delete these keys if `from_state == ERROR`.